### ISSUE TITLE
Expensive method

### ISSUE THREAD

**Issue by Daiiszuki** [jamesbachini/Uniswap-V3-Experiments#2]
Is there a less costly way to implement this? If I'm not mistaken, this needs to call a write function to get quotes

**Comment by jamesbachini** [comment]
Yes this was done pre-release. There's more documentation on it now. Take a look at this:
https://docs.uniswap.org/sdk/guides/fetching-prices
